src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2029,6 +2059,65 @@ protected override void HandleEvaluateRequestAsync(IRequestResponder<EvaluateArg
 
         }
 
+        protected override void HandleReadMemoryRequestAsync(IRequestResponder<ReadMemoryArguments, ReadMemoryResponse> responder)
+        {
+            int hr;
+            ReadMemoryArguments rma = responder.Arguments;
+            ErrorBuilder eb = new ErrorBuilder(() => AD7Resources.Error_Scenario_ReadMemory);
+            try
+            {
+                IDebugStackFrame2 frame;
+                m_frameHandles.TryGetFirst(out frame);
+                eb.CheckOutput(frame);
+
+                IDebugProperty2 property;
+                string error;
+                hr = GetIDebugPropertyFromExpression(rma.MemoryReference, frame, 0, DAPEvalFlags.NONE, out property, out error);
```

Comment:
>MemoryReference [](start = 57, length = 15)

This isn't really how the protocol is supposed to work. You are supposed to return `memoryReference` values from expressions, and this should be a format that you don't need to parse (ex: just the address string).

Reply:
BTW you might want to first test your changes by running against Visual Studio as it already fully supports this part of the protocol.

Reply:
Do we send the request in CMake Debug if we have the Memory Window Open?

Reply:
Can you tell me more about the plan for how we are going to add a memory window to VS Code? If I understand correctly, the idea is that it isn't going to really be part of VS Code, and instead we are going to somehow request memory from the adapter, save the bytes to a file, and then open up a hex editor on the bytes. But a few things I don't understand:
* How is the user going to enter the expression they want evaluated?
* How does the user specify how many bytes they want?
* How does the memory get refreshed after a step (or any other form of exiting and reentering break state)?

Reply:
In case it is useful, how I can imagine a memory window should use this feature -

1. UI sends an `evaluate` request from the user's expression, and waits for the result. If the result doesn't have a `memoryReference` - show user an error.
    A. Note: since in this case since the UI is part of the extension, maybe the UI could cheat if the expression to evaluate was just an address string and skip this step. But make sure to implement the 'real way' first so that things work in a compliant client.
2. Send a `readMemory` request with the `memoryReference` from the previous value.

Reply:
1. How is the user going to enter the expression they want evaluated?

The current idea was to make it a command. Users will use `ctrl+shift+p` and there will be a quick input where they can either type an address or a variable. Then the next input it will ask the offset then the size.

Example: https://github.com/haneefdm/cortex-debug/blob/e3faed3353e515f9802ea66fa9980043118344fc/src/frontend/extension.ts#L257

2. How does the user specify how many bytes they want?

Only interface with VS Code we can use is the Quick Input as a command.

3. How does the memory get refreshed after a step (or any other form of exiting and reentering break state)?

There will be a cache of the last request they made and on a stopping event it will refresh the page. We trace the requests using the DebugAdapterTracker.

Reply:
Thanks. In that case the description above should be correct.

Reply:
Note: I'll be working on the VS Code UI before checking this in.